libs/ml/src/ops/matrix_multiply.cpp

```diff
@@ -296,17 +296,18 @@ std::pair<OperationsCount, math::SizeVector> MatrixMultiply<T>::ChargeForward(
   assert(input_shapes.size() == 2);
 
   // Assuming this is a matrix multiplication of weights * input_vector
-  // e.g. [n; m] * [m; batch_size], then total operations cost is n * m * batch_size,
-  // and default batch_size is 1.
-  OperationsCount const n          = input_shapes.front().at(0);
-  OperationsCount const m          = input_shapes.back().at(0);
-  OperationsCount const batch_size = input_shapes.back().at(input_shapes.front().size() - 1);
+  OperationsCount const input_1_dim_1 = input_shapes.front().at(0);
+  OperationsCount const input_1_dim_2 = input_shapes.front().at(1);
+  OperationsCount const input_2_dim_1 = input_shapes.back().at(0);
+  OperationsCount const input_2_dim_2 = input_shapes.back().at(1);
+  OperationsCount const batch_size    = input_shapes.front().at(input_shapes.front().size() - 1);
```

Comment:
This is incorrect, only one matmul occurs per batch, because data are flattened (1D) and second dimmension is batch so it results into 2D tensor including all batches at once.